shell/platform/android/io/flutter/plugin/common/ErrorLogResult.java

```diff
@@ -30,11 +32,15 @@ public void success(@Nullable Object result) {}
     @Override
     public void error(String errorCode, @Nullable String errorMessage, @Nullable Object errorDetails) {
         String details = (errorDetails != null) ? " details: " + errorDetails : "";
-        Log.println(level, tag, errorMessage + details);
+        if (level >= Log.WARN || BuildConfig.DEBUG) {
+            Log.println(level, tag, errorMessage + details);
```

Comment:
I think it would be cleaner throughout to create our own utility class that wraps `android.util.Log`, has basically the same interface, but folds these build type checks into its methods automatically. That way we just need to check for people using our fork of the class instead of remembering to always check the `BuildConfig` correctly at each callsite. WDYT?